ast.hpp

```diff
@@ -1523,7 +1523,7 @@ namespace Sass {
 
       if (with)
       {
-        if (!l || !l->length()) return str.compare("rule");
+        if (!l || !l->length() != 0) return str.compare("rule") != 0;
```

Comment:
Oops. TravisCI was passing, I didn't ran tests locally. :p
Thanks for fixing it! :+1: